---
title: "Figure3_AJB_review_GWAShits_gene_distance"
output: html_notebook
---

```{r}
library(cowplot)
library(grid)
library(tidyverse)
library(scales)

data <- read.table("~/Documents/Github/AJB_MutationalTargetSize_GenomeSize/data/1001_arabidopsis_GWAS_tophit_closest_gene_distance", header=FALSE)
data$GWAS_hits_distance_gene <- abs(data$V9)

newdata <- data %>% 
  filter(GWAS_hits_distance_gene > 0)

fill <- "#4271AE"
lines <- "#1F3552"

pdf("Figure3_GWAShits_distance_gene.pdf", width=8, height=5)
arab <- ggplot(newdata, aes(x = GWAS_hits_distance_gene)) +
        geom_density(colour = lines, fill = fill,
                 size = 1) +
        scale_x_log10(name = "Arabidopsis non-genic GWAS hits
distance to nearest gene (bp, log scale)",
                      breaks=10^(0:6),
                      labels = trans_format("log10", math_format(10^.x)),
                      limits=c(1, 1000000)) +
        scale_y_continuous(name = "Density", limits=c(0, 0.8)) +
        theme_bw() +
        theme(axis.line = element_line(size=1, colour = "black"),
              panel.grid.major = element_line(colour = "#d3d3d3"),
              panel.grid.minor = element_blank(),
              panel.border = element_blank(), panel.background = element_blank(),
              plot.title = element_text(size = 18, face = "bold"),
              axis.text.x=element_text(colour="black", size = 14),
              axis.text.y=element_text(colour="black", size = 14))

arab <- arab + geom_vline(xintercept = 661, size = 1, colour = "#FF3721",
               linetype = "dashed")


### Maize use GWAS hits relative distance to AGPv2 FGS
data <- read.table("~/Documents/Github/AJB_MutationalTargetSize_GenomeSize/data/Wallace_etal_2014_GWAS_hits_AGPv2FGS_closest_gene_distance", header=FALSE)
data$GWAS_hits_distance_gene <- abs(data$V9)

newdata <- data %>% 
  filter(GWAS_hits_distance_gene > 0)

fill <- "#4271AE"
lines <- "#1F3552"

maize <- ggplot(newdata, aes(x = GWAS_hits_distance_gene)) +
        geom_density(colour = lines, fill = fill,
                 size = 1) +
        scale_x_log10(name = "Maize non-genic GWAS hits 
distance to nearest gene (bp, log scale)",
                      breaks=10^(0:6),
                      labels = trans_format("log10", math_format(10^.x)),
                      limits=c(1, 1000000)) +
        scale_y_continuous(name = "Density", limits=c(0, 0.8)) +
        theme_bw() +
        theme(axis.line = element_line(size=1, colour = "black"),
              panel.grid.major = element_line(colour = "#d3d3d3"),
              panel.grid.minor = element_blank(),
              panel.border = element_blank(), panel.background = element_blank(),
              plot.title = element_text(size = 18, face = "bold"),
              axis.text.x=element_text(colour="black", size = 14),
              axis.text.y=element_text(colour="black", size = 14))

maize <- maize + geom_vline(xintercept = 7372, size = 1, colour = "#FF3721",
               linetype = "dashed")

fill <- c("#40b8d0", "#b2d183")

GWAS_hits <- read.table("~/Documents/Github/AJB_MutationalTargetSize_GenomeSize/data/Maize_Arabidopsis_GWAS_hits.txt", header=TRUE)

GWAS_hits_distribution <- ggplot(GWAS_hits, aes(y=Percentage, x=Species, fill=Position)) +
  geom_bar(stat="identity") +
  geom_text(aes(label = paste0(Percentage,"%")),
           colour="black", size=4, position=position_stack(vjust = 0.5)) +
  theme(legend.position="bottom", legend.direction="horizontal",
        legend.title = element_blank()) +
  scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  labs(x="Species", y="Percentage") +
  scale_fill_manual(values=fill) +
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  theme(plot.title = element_text(size = 18, face = "bold"),
        axis.text.x=element_text(colour="black", size = 14),
        axis.text.y=element_text(colour="black", size = 14))

right_column <- plot_grid(arab, maize, labels = c('b', 'c'), ncol = 1, align = 'v')
plot_grid(GWAS_hits_distribution, right_column, labels = c('a', ''), ncol = 2, rel_widths = c(1, 1.3))
dev.off()
```

